### DISCUSSION TITLE
Is Termux 0.120 on Google Play the official version?

### DISCUSSION THREAD

**Discussion by 743859910** [termux/termux-app#3995] [Q&A] [answered]
![Screenshot_2024-06-07-17-48-54-830-edit_com.android.vending.png](https://github.com/termux/termux-app/assets/84552324/014ff80d-62c9-4439-9891-f5fdaf30f91c)

**Comment by fornwall** [comment, marked as answer]
It is!

The Termux app on Google Play is being revived and updated. But it's still early days, and this is still to be considered an early alpha version (it's built on a local modified fork) - will post app updates and more information next week (some things have changed due to Google Play policy requirements, and not everything has been implemented or tested yet).

So please avoid spreading the word about Google Play availability until things have stabilised.

**Reply by cherinyy** [reply]
I'd suggest you guys better update the README and other documentation, otherwise it's a surprising thing that channels marked as "deprecated" suddenly get updated.

**Reply by fornwall** [reply]
@cherinyy Thanks, will talk to people and start updating documentation next week!

**Reply by oldherl** [reply]
I was questioning whether the Google Play account has been compromised by a JiaTan-like attack.

**Reply by fornwall** [reply]
@oldherl Understandable!

Sorry for not being out earlier with information - didn't expect the approval process to go that fast, and forgot to check the checkbox to await releases after their approval.

Will post updates next week as things stabilizes and more things are clear (are in discussion with a reviewer if more functionality that now have been removed due to google play policy requirements can be added back).

**Reply by vivlim** [reply]
Yeah, please consider at *least* updating the readme to mention that this version exists and is an alpha, (maybe linking to this thread) I nearly filed an issue but then happened to check the discussions page.

By the way, I had installed Termux from a github release, but play store still offered 0.120 to me as an update. You might see significant traffic if that happens for other people too.

**Reply by Dhyfer1** [reply]
@fornwall can we see Termux with support for Monet (Material You) from Android 12 onwards?

**Reply by alandenniseaton** [reply]
Why does Google Play think that I have this installed?
I have the F-Droid version. I do not want Google Play notifying me of updates to the Play version or inserting it into my update list. That will get very annoying very quickly.

How do I fix this?

**Reply by tmzullinger** [reply]
It is hard for folks to not notice the Play Store build since even if they have installed from F-Droid (or Github, AFAIK), the Play Store offers this update -- which it can't actually install over those versions.

Any chance of this new build being moved to the Play Store beta channel so that interested testers can opt in but most other folks won't have an update they can't install in their Play Store update queue for however long it may take to go from alpha to beta to stable release @fornwall?

**Reply by sanko** [reply]
The package name (`com.termux`) would need to be changed. Install will fail because the signatures don't match which it actually more annoying than if it worked properly. Flagging it as beta is only a stopgap.

**Reply by Salientekill** [reply]
Currently it does not have storage permissions, and it is closing automatically when I use termux-setup-storage, Also add notifications options please

**Reply by fornwall** [reply]
@Salientekill Thanks for the feedback! The `termux-setup-storage` crash will be fixed shortly for internal storage (in app version 0.122, probably released tomorrow if Google Play review works as expected), and notifications will likely come within a week. Check out for https://github.com/termux-play-store for current status and upates.

For further feedback and issues, you're welcome to report issues at https://github.com/termux-play-store/termux-issues/issues/new/choose!

**Comment by charnysh** [comment]
I have Termux version 0.118.0 installed and plugins from the F-Droid repository.  Today Google Play prompted me to update the application to version 0.120.  As far as I understand from the explanation in the README, I am not allowed to mix versions of the application from different sources, so I did not update and turned off automatic updating for this application in Google Play.  Did I do the right thing?
Will there be version 0.120 on F-Droid and why is it not in the releases on GitHub?
Will there be support for Android versions lower than 11, such as Android 7, and how will this be implemented?
Thank you!

**Reply by fornwall** [reply]
> I did not update and turned off automatic updating for this application. Did I do the right thing?

@viccher Yes, you did the right thing, it's better to stay on the F-Droid version for now!

> Will there be version 0.120 on F-Droid and why is it not in the releases on GitHub?

The version on Google Play is currently a modified version of the app, which some functionality modified and some functionality adopted or removed. In the long run many (most/all?) changes will probably be merged to the generic open source distribution of Termux, but it will take some time and exploration.

> Will there be support for Android versions lower than 11, such as Android 7, and how will this be implemented?

Probably in the long run, but for now Termux on Google Play is restricted to Android 11 for simplicity.

**Reply by charnysh** [reply]
Thanks again for your answer and best wishes!  :)

**Reply by luisdavim** [reply]
What are the main differences on the play store version? Any major limitations? Any new functionality or bug fixes?

**Reply by fornwall** [reply]
@luisdavim I wrote up https://github.com/termux-play-store with some summary about that!

**Reply by luisdavim** [reply]
Thanks, why was the fork done from an older version? It seems to be about 10 releases behind...

**Reply by fornwall** [reply]
> Thanks, why was the fork done from an older version? It seems to be about 10 releases behind...

It wasn't - but some recent functionality has not been ported yet (either because of time constraints, or permissions not yet accepted by Google Play reviewers), and some things have been done differently due to having access to newer Android API:s (for instance [Insets: Apply rounded corners](https://developer.android.com/develop/ui/views/layout/insets/rounded-corners)). If you are interested you are very welcome to try it out and [report issues](https://github.com/termux-play-store/termux-issues/issues/new/choose) if you discover anything specifically missing or broken!

**Comment by charnysh** [comment]
I think it would be correct in the current situation to exclude the possibility of updating in Google Play versions of the application installed from other sources (F-Droid and GitHub).

**Reply by fornwall** [reply]
@viccher Good point! I think bumping the version number (to say 0.300.0) on F-Droid would be a good solution for now, so it has a higher version than on Google Play. I'll create a PR for that later today when I'm in front of a computer.

**Comment by agnostic-apollo** [comment]
An official response regarding the Google PlayStore app updates including `termux-app` `v0.120` has been made at https://github.com/termux/termux-app/discussions/4000
